akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -239,6 +239,63 @@ private[akka] final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out, de
 /**
  * INTERNAL API
  */
+private[akka] final case class Intersperse[T](start: Option[T], inject: T, end: Option[T]) extends StatefulStage[T, T] {
```

Comment:
Made it a ticket https://github.com/akka/akka/issues/18736 want to push through json streaming today, can come back optimising this stage after